spot_wrapper/tests/test_wrapper.py

```diff
@@ -120,3 +122,144 @@ def test_spot_wrapper_commands(simple_spot: SpotFixture, simple_spot_wrapper: Sp
+    assert_non_processing_stand_status(simple_spot_wrapper.is_standing)
+
+    # initial sitting state
+    assert simple_spot_wrapper.last_sit_command is None
+    assert not simple_spot_wrapper.is_sitting
+
+    # sit command with non-processing status - should reset the command and leave state unchanged
+    assert_non_processing_sit_status(simple_spot_wrapper.is_sitting)
+
+    # sit command with processing status
+    simple_spot_wrapper.sit()
+    assert simple_spot_wrapper.last_sit_command is not None
+    assert not simple_spot_wrapper.is_sitting
+    response = RobotCommandFeedbackResponse()
+    response.feedback.synchronized_feedback.mobility_command_feedback.status = (
+        RobotCommandFeedbackStatus.STATUS_PROCESSING
+    )
+    # command in progress
+    response.feedback.synchronized_feedback.mobility_command_feedback.sit_feedback.status = (
+        SitCommand.Feedback.STATUS_IN_PROGRESS
+    )
+    update_with_feedback_response(response)
+    assert simple_spot_wrapper.last_sit_command is not None
+    assert not simple_spot_wrapper.is_sitting
+    assert not simple_spot_wrapper.is_standing
+    # command finished
+    response.feedback.synchronized_feedback.mobility_command_feedback.sit_feedback.status = (
+        SitCommand.Feedback.STATUS_IS_SITTING
+    )
+    update_with_feedback_response(response)
+    assert simple_spot_wrapper.last_sit_command is None
+    assert simple_spot_wrapper.is_sitting
+    assert not simple_spot_wrapper.is_standing
+
+    # sit command with non-processing status - should reset the command and leave state unchanged
+    assert_non_processing_sit_status(simple_spot_wrapper.is_sitting)
+
+
+def test_robot_movement_states(simple_spot: SpotFixture, simple_spot_wrapper: SpotWrapper) -> None:
+    # bosdyn.api.RobotCommandService/RobotCommand implementation is mocked
+    # (as spot_wrapper.testing.mocks.MockSpot is automatically specified),
+    # so here we provide the same response for every future command.
+    response = RobotCommandResponse()
+    response.status = RobotCommandResponse.Status.STATUS_OK
+    simple_spot.api.RobotCommand.future.returns(response).forever()
+
+    assert simple_spot_wrapper.last_sit_command is None
+    assert simple_spot_wrapper.last_stand_command is None
+    assert simple_spot_wrapper.last_velocity_command_time is None
+    assert simple_spot_wrapper.is_sitting
+    assert not simple_spot_wrapper.is_standing
+    assert not simple_spot_wrapper.is_moving
+
+    simple_spot_wrapper.velocity_cmd(0, 0, 0)
+    assert simple_spot_wrapper.last_velocity_command_time is not None
+    simple_spot_wrapper.updateTasks()
+    time.sleep(0.1)
+
+    assert simple_spot_wrapper.last_sit_command is None
+    assert simple_spot_wrapper.last_stand_command is None
+    assert not simple_spot_wrapper.is_sitting
+    assert simple_spot_wrapper.is_standing
+    assert simple_spot_wrapper.is_moving
```

Comment:
Question: shouldn't a velocity command of 0 not stand up+move the robot in practice?

Reply:
From what I have tested practically with the robot, once it starts receiving any velocity commands (zero included) it will always stand up and override any following sit/stand commands.

Logically, I think it does make sense (in relation to the intended behavior discussed here https://github.com/bdaiinstitute/spot_ros2/issues/175), because any valid velocity command implies that the robot should to be ready to move, and a zero velocity simply means standing in place.

Reply:
Ok, makes sense!